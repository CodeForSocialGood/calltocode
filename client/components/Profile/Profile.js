import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import ProjectActionCreator from '../../actions/project'
import UserActionCreator from '../../actions/user'
import ProfilePictureForm from './ProfilePictureForm/ProfilePictureForm'
import ListOfProjects from '../ListOfProjects/ListOfProjects'
import styles from './Profile.scss'

class Profile extends Component {
  componentDidMount () {
    this.props.onLoad(this.props.user)
  }

  render () {
    const title =
      this.props.user.usertype === 'contact'
        ? "Your Organization's Projects"
        : 'Projects Applied For'

    return (
      <div className={styles.profilePage}>
        <div className={styles.profilePictureForm}>
          <ProfilePictureForm
            user={this.props.user}
            updateProfilePicture={this.props.updateProfilePicture} />
        </div>

        <div className={styles.listOfProjects}>
          <h3 className={styles.youProjects}>Your Projects</h3>
          <ListOfProjects title={title} projects={this.props.projects} />
        </div>
      </div>
    )
  }
}

function mapStateToProps (state) {
  return {
    projects: state.project.projects,
    user: state.user
  }
}

const mapDispatchToProps = {
  onLoad: ProjectActionCreator.fetchProfileProjects,
  updateProfilePicture: UserActionCreator.updateProfilePicture
}

Profile.propTypes = {
  onLoad: PropTypes.func,
  updateProfilePicture: PropTypes.func,
  projects: PropTypes.array,
  user: PropTypes.object
}

export default connect(mapStateToProps, mapDispatchToProps)(Profile)
