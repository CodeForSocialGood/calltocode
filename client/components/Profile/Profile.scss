.profilePage {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: auto;
  grid-row-gap: 15px;
  padding: 0 60px;
}

.profilePictureForm {
  grid-column: 1 / 2;
  grid-row: 1 / 2;
}

.listOfProjects {
  grid-column: 2 / 4;
  grid-row: 1 / 4;
}
