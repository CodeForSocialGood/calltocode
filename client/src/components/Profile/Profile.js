import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import ProjectActionCreator from '../../actions/project'
import ListOfProjects from '../ListOfProjects/ListOfProjects'

class Profile extends Component {
  componentDidMount () {
    this.props.getProfileOpps(this.props.user)
  }

  render () {
    const title = this.props.user.usertype === 'contact'
      ? 'Your Organization\'s Projects'
      : 'Projects Applied For'

    return (
      <ListOfProjects
        title={title}
        projects={this.props.projects} />
    )
  }
}

function mapStateToProps (state) {
  return {
    projects: state.project.projects,
    user: state.user
  }
}

const mapDispatchToProps = {
  getProfileOpps: ProjectActionCreator.getProfileOpps
}

Profile.propTypes = {
  projects: PropTypes.array,
  getProfileOpps: PropTypes.func,
  user: PropTypes.object
}

export default connect(mapStateToProps, mapDispatchToProps)(Profile)
