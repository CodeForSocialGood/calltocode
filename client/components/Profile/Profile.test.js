/* eslint-env jest */
import React from 'react'
import {shallow} from 'enzyme'

import {Profile} from './Profile'
let mockProps

beforeEach(() => {
  mockProps = {
    onLoad: () => {},
    projects: [],
    user: {
      usertype: null
    },
    mockState: {
      fetchProfileProject: () => {}
    }
  }
})

describe('<Profile />', () => {
  describe('user\'s usertype is "contact"', () => {
    beforeEach(() => { mockProps.user.usertype = 'contact' })

    test('component renders correctly', () => {
      const component = shallow(<Profile {...mockProps} />)
      expect(component).toMatchSnapshot()
    })
  })
  describe('user\'s usertype is not "contact"', () => {
    beforeEach(() => { mockProps.user.usertype = '' })

    test('component renders correctly', () => {
      const component = shallow(<Profile {...mockProps} />)
      expect(component).toMatchSnapshot()
    })
  })
})
