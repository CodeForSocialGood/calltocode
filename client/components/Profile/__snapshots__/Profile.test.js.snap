// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> user's usertype is "contact" component renders correctly 1`] = `
<Connect(ListOfProjects)
  projects={Array []}
  title="Your Organization's Projects"
/>
`;

exports[`<Profile /> user's usertype is not "contact" component renders correctly 1`] = `
<Connect(ListOfProjects)
  projects={Array []}
  title="Projects Applied For"
/>
`;
